src/axom/primal/geometry/Polygon.hpp

```diff
@@ -123,6 +91,45 @@ class Polygon
   AXOM_HOST_DEVICE Polygon()
   { }
 
+  /*!
+   * \brief Destructor for Polygon. Suppress CUDA warnings for
+   *        dynamic axom::Array.
+   */
+  AXOM_SUPPRESS_HD_WARN
+  AXOM_HOST_DEVICE
+  ~Polygon() { m_vertices.clear(); }
```

Comment:
To get rid of CUDA __host__ __device__ warnings when compiling for the `axom::Array` option, I added `AXOM_SUPPRESS_HD_WARN` here in the Polygon class, as well as to the `clip(Polygon, Polygon)` overload.

I would rather not do this.

Is there a way I can refactor/reorganize the current implementation  to better clear out warnings?

Reply:
@bmhan12 are the warnings about calling a device function in host code?

Reply:
What happens if you call `clear()` instead of `m_vertices.clear()`?  It looks like you added specialization of the clear method to avoid the warnings.

Reply:
Same warnings. The warnings are coming from `clip_impl.hpp`, when the clipPolygonPolygon() function is compiled for device.

Reply:
You might be able to move the underlying storage into a templatized base class, that's specialized for `axom::Array` vs `StaticArray`. In the `axom::Array` specialization, you could then define the destructor as host-only, and only have `AXOM_HOST_DEVICE` on the `StaticArray` specialization's destructor.

Reply:
@publixsubfan Thanks for the suggestion. I gave it a try, but still needed the `AXOM_SUPPRESS_HD_WARN `macro to suppress warnings.